driver-kotlin-sync/src/main/kotlin/com/mongodb/kotlin/client/DistinctIterable.kt

```diff
@@ -99,4 +100,20 @@ public class DistinctIterable<T : Any?>(private val wrapped: JDistinctIterable<T
      * @return this
      */
     public fun comment(comment: BsonValue?): DistinctIterable<T> = apply { wrapped.comment(comment) }
+
+    /**
+     * Sets the hint for which index to use. A null value means no hint is set.
+     *
+     * @param hint the hint
+     * @return this
+     */
+    public fun hint(@Nullable hint: Bson?): DistinctIterable<T> = apply { wrapped.hint(hint) }
+
+    /**
+     * Sets the hint for which index to use. A null value means no hint is set.
+     *
+     * @param hint the name of the index which should be used for the operation
+     * @return this
+     */
+    public fun hintString(@Nullable hint: String?): DistinctIterable<T> = apply { wrapped.hintString(hint) }
```

Comment:
again, could we add the following detail to above to be consistent with `FindIterable.kt`?
```
     * Note: If [DistinctIterable.hint] is set that will be used instead of any hint string.
```